lib/mix/lib/mix/dep.ex

```diff
@@ -167,10 +167,10 @@ defmodule Mix.Dep do
     do: "lock mismatch: the dependency is out of date (run \"mix deps.get\" to fetch locked version)"
```

Comment:
We don't know why there is a lock mismatch. Your reported case is just one of the instances when you can get this message.